### DISCUSSION TITLE
Plans for rendering markdown with the assistant tab?

### DISCUSSION THREAD

**Discussion by creda-technologies** [zed-industries/zed#8257] [Feature Requests]
I'm really enjoying using the assistants tab with my OpenAI API keys, however, with responses that usually revolve around helping me generate code (in my case, pydantic classes that map to a submitted document), I am wondering if there is any plan to support markdown rendering similar to the ChatGPT app itself. For example, generating the code blocks, bolding and enlarging titles, that sort of thing.

**Comment by Moshyfawn** [comment]
From what I saw, I don't think anyone has ever mentioned the markdown preview in the assistant panel. However, Zed does have a `markdown: open preview` option for markdown files, so it might be possible to add a toggle for markdown preview for assistant code blocks.

It's worth noting that the assistant panel already supports markdown syntax highlighting.

Before proposing this feature request, I would double check if it has already been brought up. If not, go ahead and propose it in the issues!

**Reply by creda-technologies** [reply]
I see that now, that assistant has basic rendering and code highlighting for code blocks which is great. But I was wondering if there are plans to make it more like the chatGPT app, where code blocks are rendered in these little boxes with a copy button? But I do now see that there is syntax highlighting within the panel.

**Reply by 10bn** [reply]
+1

**Reply by electic** [reply]
I just found Zed and love it. @Moshyfawn I think having markdown previews boxes like GPT would just be amazing.

**Comment by dandv** [comment]
[As of Dec 2024](https://github.com/zed-industries/zed/issues/20724#issuecomment-2517607031),

> There is active development work to restyle the Assistant Panel, including support for displaying code blocks

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
